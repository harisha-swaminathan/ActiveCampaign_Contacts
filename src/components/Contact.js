import React, { useState, useEffect } from 'react';
import axios from 'axios';

const Contact = (contact) => {
  const { id, firstName, lastName } = contact.contact;
  const [name, setName] = useState('');
  const [deals, setDeals] = useState(-1);
  const [location, setLocation] = useState(null);
  const [, setError] = useState('');
  const [tags, setTags] = useState([]);

  const isLoaded = () => {
    const loaded = !!name && deals > -1 && typeof location === 'string' && tags.length > 0;
    return loaded;
  };

  useEffect(() => {
    (async () => {
      try {
        const response = await axios.get(`https://cors-anywhere.herokuapp.com/https://sahmed93846.api-us1.com/api/3/contacts/${id}`, {
          headers: {
            'Api-Token': 'bcd062dedabcd0f1ac8a568cdcf58660c44d7e79b91763cc1a5d0c03d52c522d851fceb0',
          },
        });
        const tagResponse = await axios.get(`https://cors-anywhere.herokuapp.com/https://sahmed93846.api-us1.com/api/3/contacts/${id}/contactTags`, {
          headers: {
            'Api-Token': 'bcd062dedabcd0f1ac8a568cdcf58660c44d7e79b91763cc1a5d0c03d52c522d851fceb0',
          },
        });
        await tagResponse.data.contactTags.forEach(async (contactTag) => {
          const tag = await axios.get(`https://cors-anywhere.herokuapp.com/${contactTag.links.tag}`, {
            headers: {
              'Api-Token': 'bcd062dedabcd0f1ac8a568cdcf58660c44d7e79b91763cc1a5d0c03d52c522d851fceb0',
            },
          });
          setTags((prevTags) => [...prevTags, tag.data.tag.tag]);
        });
        const geoAddress = response.data.geoAddresses ? `${response.data.geoAddresses[0].city}, ${response.data.geoAddresses[0].state}, ${response.data.geoAddresses[0].country}` : ' ';
        const totalDeals = response.data.deals ? response.data.deals.length : 0;
        setName(`${firstName} ${lastName}`);
        setDeals(totalDeals);
        setLocation(geoAddress);
      } catch (error) {
        setError(error.message);
      }
    })();
  }, []);
  return (
    <>
      { isLoaded()
               && (
               <>
                 <td><input type="checkbox" /></td>
                 <td>
                   <img src="/man.png" alt="avatar" className="table__avatar" />
                   {name}
                 </td>
                 <td>$100</td>
                 <td>{location}</td>
                 <td>{deals}</td>
                 <td>
                   {tags.map((tag, i) => (
                     // eslint-disable-next-line react/no-array-index-key
                     <span key={i}>
                       {tag}
                       ,
                       {' '}
                     </span>
                   ))}
                 </td>

               </>
               )}
    </>
  );
};

export default Contact;
