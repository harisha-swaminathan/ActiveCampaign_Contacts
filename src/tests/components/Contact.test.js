import React from 'react';
import { mount } from 'enzyme';
import axios from 'axios';
import contact from '../fixtures/contact';
import Contact from '../../components/Contact';

let wrapper;
describe('Render component and fetch from api', () => {
  beforeEach(() => {
    jest.mock('axios');
    axios.get = jest.fn();
    axios.get.mockImplementationOnce(() => Promise.resolve(contact));
    wrapper = mount(<Contact key={1} contact={contact.contact} />);
  });

  test('should render Contact correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
