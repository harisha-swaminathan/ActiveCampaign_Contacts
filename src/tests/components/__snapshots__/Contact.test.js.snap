// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component and fetch from api should render Contact correctly 1`] = `
<Contact
  contact={
    Object {
      "accountContacts": Array [
        "1",
      ],
      "adate": null,
      "bounced_date": null,
      "bounced_hard": "0",
      "bounced_soft": "0",
      "cdate": "2007-05-05T12:49:09-05:00",
      "contactAutomations": Array [
        "1",
      ],
      "contactLists": Array [
        "1",
      ],
      "deals": Array [
        "1",
      ],
      "deleted": "0",
      "edate": null,
      "email": "selmer.koss@example.com",
      "email_domain": "",
      "email_local": "",
      "fieldValues": Array [
        "1",
      ],
      "firstName": "Charles",
      "geoIps": Array [
        "1",
      ],
      "gravatar": "0",
      "hash": "",
      "id": "1",
      "ip": "0",
      "lastName": "Reynolds",
      "links": Object {
        "bounceLogs": "/1/bounceLogs",
        "contactAutomations": "/1/contactAutomations",
        "contactData": "/1/contactData",
        "contactDeals": "/1/contactDeals",
        "contactGoals": "/1/contactGoals",
        "contactLists": "/1/contactLists",
        "contactLogs": "/1/contactLogs",
        "contactTags": "/1/contactTags",
        "deals": "/1/deals",
        "fieldValues": "/1/fieldValues",
        "geoIps": "/1/geoIps",
        "notes": "/1/notes",
        "organization": "/1/organization",
        "plusAppend": "/1/plusAppend",
        "scoreValues": "/1/scoreValues",
        "trackingLogs": "/1/trackingLogs",
      },
      "organization": null,
      "orgid": "0",
      "phone": "",
      "rating_tstamp": null,
      "segmentio_id": "",
      "sentcnt": "0",
      "socialdata_lastcheck": null,
      "ua": null,
      "udate": null,
    }
  }
>
  <td
    align="center"
    colSpan="6"
  >
    Loading...
  </td>
</Contact>
`;
